src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java

```diff
@@ -137,61 +141,45 @@ public List<ModificationInfos> getModifications(UUID groupUuid, boolean onlyMeta
                     .collect(Collectors.toList());
         }
     }
 
-    public TabularModificationEntity loadTabularModificationSubEntities(ModificationEntity modificationEntity) {
-        TabularModificationEntity tabularModificationEntity = (TabularModificationEntity) modificationEntity;
-        switch (tabularModificationEntity.getModificationType()) {
-            case GENERATOR_MODIFICATION:
-                tabularModificationEntity = modificationRepository.findAllWithReactiveCapabilityCurvePointsById(modificationEntity.getId()).orElseThrow(() ->
-                        new NetworkModificationException(MODIFICATION_NOT_FOUND, String.format(MODIFICATION_NOT_FOUND_MESSAGE, modificationEntity.getId()))
-                );
-                modificationRepository.findAllReactiveCapabilityCurvePointsByIdIn(tabularModificationEntity.getModifications().stream().map(ModificationEntity::getId).toList());
-                break;
-            default:
-                break;
-        }
-        return tabularModificationEntity;
-    }
-
-    public ModificationInfos getModificationInfos(ModificationEntity modificationEntity) {
-        if (modificationEntity instanceof TabularModificationEntity) {
-            return loadTabularModificationSubEntities(modificationEntity).toModificationInfos();
-        }
-        return modificationEntity.toModificationInfos();
-    }
-
     public List<ModificationInfos> getModificationsInfos(List<UUID> groupUuids, boolean onlyStashed) {
-        Stream<ModificationEntity> modificationEntity = groupUuids.stream().flatMap(this::getModificationEntityStream);
+        Stream<ModificationEntity> modificationEntities = groupUuids.stream().flatMap(this::getModificationEntityStream);
         if (onlyStashed) {
-            return modificationEntity.filter(m -> m.getStashed() == onlyStashed)
+            return modificationEntities.filter(m -> m.getStashed() == onlyStashed)
                     .map(this::getModificationInfos)
                     .collect(Collectors.toList());
         } else {
-            return modificationEntity.map(this::getModificationInfos)
+            return modificationEntities.map(this::getModificationInfos)
                 .collect(Collectors.toList());
         }
     }
 
     @Transactional(readOnly = true)
-    public ModificationInfos getModificationInfo(UUID modificationUuid) {
-        Optional<ModificationEntity> optionalModificationEntity = modificationRepository.findById(modificationUuid);
-        if (!optionalModificationEntity.isPresent()) {
-            throw new NetworkModificationException(MODIFICATION_NOT_FOUND, modificationUuid.toString());
+    public ModificationInfos getModificationInfos(UUID modificationUuid) {
+        return modificationRepository.findById(modificationUuid)
+            .map(this::getModificationInfos)
+            .orElseThrow(() -> new NetworkModificationException(MODIFICATION_NOT_FOUND, modificationUuid.toString()));
+    }
```

Comment:
IMO, should be as following to avoid throw the same exception when getModificationInfos return a null value and keep the same behavior as the original implementation

public ModificationInfos getModificationInfos(UUID modificationUuid) {
        ModificationEntity modificationEntity = modificationRepository.findById(modificationUuid)
            .orElseThrow(() -> new NetworkModificationException(MODIFICATION_NOT_FOUND, modificationUuid.toString()));

        return  getModificationInfos(modificationEntity);
}